### PR TITLE
Automated fix for two controls:

### PR DESCRIPTION
**Pull request by iamPallav**
## Description
</br>
Below are the modified controls:

-ADO_Environment_AuthZ_Dont_Grant_All_Pipelines_Access-commit 1
- ADO_Environment_AuthZ_Restrict_Broader_Group_Access-commit 1
-  ADO_Release_SI_Review_URL_Variables_Settable_At_Release_Time - commit 2
- ADO_Build_SI_Review_URL_Variables_Settable_At_Queue_Time- commit 3

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1 [hunk 1]
```diff
@@ -880,6 +884,47 @@ class CommonSVTControls: ADOSVTBase {
...
+                $RawDataObjForControlFix.allPipelines.authorizedOn = $null;
+                $body = $RawDataObjForControlFix | ConvertTo-Json -Depth 10;
+                $projectId = ($this.ResourceContext.ResourceId -split "project/")[-1].Split('/')[0]
+                $uri = "https://dev.azure.com/{0}/{1}/_apis/pipelines/pipelinePermissions/environment/{2}?api-version=5.1-preview.1" -f ($this.OrganizationContext.OrganizationName), $projectId, $($this.ResourceContext.ResourceDetails.id);
+               
+
+                $header = [WebRequestHelper]::GetAuthHeaderFromUriPatch($uri)
+                $Result = Invoke-RestMethod -Uri $uri -Method Patch -ContentType "application/json" -Headers $header -Body $body
+            
+                $controlResult.AddMessage([VerificationResult]::Fixed,  "Service connection is not accessible to all YAML pipelines.");
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1:907]: Service Connection -> Environment

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1 [hunk 2]
```diff
@@ -880,6 +884,47 @@ class CommonSVTControls: ADOSVTBase {
...
+            }
+            else {
+                $body = $RawDataObjForControlFix | ConvertTo-Json -Depth 10;
+                $projectId = ($this.ResourceContext.ResourceId -split "project/")[-1].Split('/')[0]
+                $uri = "https://dev.azure.com/{0}/{1}/_apis/pipelines/pipelinePermissions/environment/{2}?api-version=5.1-preview.1" -f ($this.OrganizationContext.OrganizationName), $projectId, $($this.ResourceContext.ResourceDetails.id);
+                $header = [WebRequestHelper]::GetAuthHeaderFromUriPatch($uri)
+                $Result = Invoke-RestMethod -Uri $uri -Method Patch -ContentType "application/json" -Headers $header -Body $body
+            
+                $controlResult.AddMessage([VerificationResult]::Fixed,  "Service connection is accessible to all YAML pipelines.");
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1:916]: Service Connection -> Environment

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1 [hunk 3]
```diff
@@ -827,12 +827,13 @@ class Build: ADOSVTBase
 
     hidden [ControlResult] CheckSettableAtQueueTimeForURL([ControlResult] $controlResult)
     {
-        $controlResult.VerificationResult = [VerificationResult]::Verify
+        $controlResult.VerificationResult = [VerificationResult]::Failed
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1:830]: Not sure if we should change the result to failed. Similarly for release as well.

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1 [hunk 4]
```diff
@@ -877,6 +887,48 @@ class Build: ADOSVTBase
...
+            $RawDataObjForControlFix = ([ControlHelper]::ControlFixBackup | where-object {$_.ResourceId -eq $this.ResourceId}).DataObject
+            
+            $uri = "https://dev.azure.com/{0}/{1}/_apis/build/definitions/{2}?api-version=5.0-preview.6" -f ($this.OrganizationContext.OrganizationName), $($this.BuildObj.project.id), $($this.BuildObj.id) 
+            $header = [WebRequestHelper]::GetAuthHeaderFromUriPatch($uri)
+            if (-not $this.UndoFix) {
+                $RawDataObjForControlFix | ForEach-Object {
+                if ([Helpers]::CheckMember($this.BuildObj[0].variables.$($_.Name), "allowOverride")  ){ $this.BuildObj[0].variables.$($_.Name).allowOverride = $false;}  }
+                $body = $this.BuildObj[0] | ConvertTo-Json -Depth 10
+                $buildDefnsObj = Invoke-RestMethod -Uri $uri -Method PUT -ContentType "application/json" -Headers $header -Body $body
+                $controlResult.AddMessage([VerificationResult]::Fixed,"Pipeline variables unmarked settable at queue time and containing URLs.");
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1:902]: We should display the list of variables that have been fixed. The following pipeline variables have been unmarked....: <insert list of vars here>. Similar for release.

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1 (hunk 3 above)
Comment (iamPallav) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1]: Made changes in PartialManager.ps1 to capture the backup of those who are verify and failed , now onwards verify will also work.

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1 (hunk 4 above)
Comment (iamPallav) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1]: Done

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1 [hunk 5]
```diff
@@ -877,6 +887,52 @@ class Build: ADOSVTBase
...
+                if ([Helpers]::CheckMember($this.BuildObj[0].variables.$($_.Name), "allowOverride")  ){ $this.BuildObj[0].variables.$($_.Name).allowOverride = $false;}  }
+                $body = $this.BuildObj[0] | ConvertTo-Json -Depth 10
+                $buildDefnsObj = Invoke-RestMethod -Uri $uri -Method PUT -ContentType "application/json" -Headers $header -Body $body
+                $controlResult.AddMessage([VerificationResult]::Fixed,"The following pipeline variables unmarked settable at queue time and containing URLs :");
+                $display = ($RawDataObjForControlFix.Name |  FT -AutoSize | Out-String -Width 512)
+                $controlResult.AddMessage("$display"); 
+            }
+            else {
+                $allVars = Get-Member -InputObject $this.BuildObj[0].variables -MemberType Properties
+                $allVars | ForEach-Object {
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1:908]: undofix is not working in case of multiple variables in backup

**Review by juhi037 — APPROVED** [review]

**Review by SaikumarRagala — APPROVED** [review]

**Pull request merged**